src/mondemand_backend_stats_rrd_builder.erl

```diff
@@ -87,43 +88,61 @@ calculate_context (Prefix, FileKey, Dirs) ->
   }.
 
 build (BuilderContext) ->
-  gen_server:cast (?NAME, {build, BuilderContext}).
+  mondemand_backend_worker_pool_sup:process (mdbes_rrd_builder_pool, { udp, 0, 0, 0, BuilderContext } ).
 
 %%====================================================================
-%% gen_server callbacks
+%% mondemand_backend_worker callbacks
 %%====================================================================
-init ([]) ->
-  { ok, #state {} }.
-
-handle_call (Request, From, State = #state {}) ->
-  error_logger:warning_msg ("~p : Unrecognized call ~p from ~p~n",
-                            [?MODULE, Request, From]),
-  { reply, ok, State }.
-
-handle_cast ({ build,
-               Context = #mdbes_rrd_builder_context { file_key = FileKey }
-             },
-             State = #state {}) ->
-  case maybe_create_files (Context) of
+create (Config) ->
+  ConnectionConfig =
+    case proplists:get_value (path, Config, undefined) of
+      undefined ->
+        Host = proplists:get_value (host, Config, "127.0.0.1"),
+        Port = proplists:get_value (port, Config, 11211),
+        {Host, Port};
+      Path ->
+        Path
+    end,
+  ConnectTimeout = proplists:get_value (connect_timeout, Config, 1000),
+  SendTimeout = proplists:get_value (send_timeout, Config, 100),
+  RecvTimeout = proplists:get_value (recv_timeout, Config, 50),
+
+  {ok, #state { connection_config = ConnectionConfig,
+                connect_timeout = ConnectTimeout,
+                send_timeout = SendTimeout,
+                recv_timeout = RecvTimeout,
+                connection = undefined
+              }}.
+
+connected (#state { connection = undefined }) -> false;
+connected (_) -> true.
+
+connect (State = #state { connection_config = ConnectionConfig,
+                          connect_timeout = ConnectTimeout,
+                          send_timeout = SendTimeout,
+                          recv_timeout = RecvTimeout
+                        }) ->
+  case rrdcached_client:open (ConnectionConfig, ConnectTimeout,
+                              SendTimeout, RecvTimeout) of
+    {ok, Client} ->
+      {ok, State#state { connection = Client }};
+    Error ->
+      {Error, State}
+  end.
+
+send (State = #state {connection = Client0}, {udp, _, _, _, 
```

Comment:
Please add a comment here maybe with a TODO that this is a hack to get the pool to pass this through and that it requires the pass_raw_data = true in the config in order to work.